packages/runtime/runtime-utils/src/test/serializer.spec.ts

```diff
@@ -22,14 +22,35 @@ describe("FluidSerializer", () => {
         });
     }
 
+    function createNestedCases(testCases: any[]) {
+        // Add an object where each field references one of the JSON serializable types.
+        testCases.push(
+            testCases.reduce<any>(
+                (o, value, index) => {
+                    o[`f${index}`] = value;
+                    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+                    return o;
+                },
+                {}));
+
+        // Add an array that contains each of our constructed test cases.
+        testCases.push([...testCases]);
+
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+        return testCases;
+    }
+
     describe("vanilla JSON", () => {
         const context = new MockHandleContext();
         const serializer = new FluidSerializer(context);
         const handle = new RemoteFluidObjectHandle("/root", context);
 
-        // Start with the various JSON-serializable types
+        // Start with the various JSON-serializable types.  A mix of "truthy" and "falsy" values
+        // are of particular interest.
+
         // eslint-disable-next-line no-null/no-null
-        const simple = [true, 1, "x", null, [], {}];
+        const simple = createNestedCases([false, true, 0, 1, "", "x", null, [], {}]);
```

Comment:
Added some "falsy" values as a sanity check that the 'value && ...' and '!!value && ...' logic above is behaving correctly.